agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java

```diff
@@ -876,31 +877,77 @@ static boolean applyCommonTags(
     if (applyConnectionStringAndRoleNameOverrides(telemetryBuilder, value, key)) {
       return true;
     }
-    if (key.equals(AiSemanticAttributes.ROLE_INSTANCE_ID.getKey()) && value instanceof String) {
-      telemetryBuilder.addTag(ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString(), (String) value);
-      return true;
-    }
-    if (key.equals(AiSemanticAttributes.APPLICATION_VERSION.getKey()) && value instanceof String) {
+    if (key.equals(AiSemanticAttributes.PREVIEW_APPLICATION_VERSION.getKey())
+        && value instanceof String) {
       telemetryBuilder.addTag(ContextTagKeys.AI_APPLICATION_VER.toString(), (String) value);
       return true;
     }
     return false;
   }
 
+  private static final WarningLogger connectionStringAttributeNoLongerSupported =
```

Comment:
can we reuse a single WarningLogger and simply update its message for different config? Feel not really needed.

Reply:
ya I think you are right, but a bit tricky to get the warning message output right, so I created work item to revisit sometime after the release: https://msazure.visualstudio.com/One/_workitems/edit/15574565